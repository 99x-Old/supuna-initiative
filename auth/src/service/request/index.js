import axios from 'axios';
import type { HeaderType, ResponseType } from 'type/response.type';
import Response from './response';

class Request {
    contentType: string;

    headers: [];

    setDefault() {
      const service = axios.create({
        headers: this.getHeaders(),
      });
      service.interceptors.response.use(Response);
      this.service = service;
      this.contentType = 'application/x-www-form-urlencoded;charset=utf-8';
    }

    setHeaders(headers: []): Request {
      this.headers = headers;
      return this;
    }

    setContentType(contentType: string): Request {
      this.contentType = contentType;
      return this;
    }

    getHeaders(): HeaderType {
      return {
        ...{
          'Access-Control-Allow-Origin': '*',
          'Content-Type': this.contentType,
        },
        ...this.headers,
      };
    }

    get(path: string, param: any): Promise<ResponseType> {
      this.setDefault();
      return this.service.get(path, param);
    }

    post(path: string, payload: any): Promise<ResponseType> {
      this.setDefault();
      return this.service.request({
        method: 'POST',
        url: path,
        responseType: 'json',
        data: payload,
      });
    }

    put(path: string, payload: any): Promise<ResponseType> {
      this.setDefault();
      return this.service.request({
        method: 'PUT',
        url: path,
        responseType: 'json',
        data: payload,
      });
    }

    delete(path: string, payload: any): Promise<ResponseType> {
      this.setDefault();
      return this.service.request({
        method: 'DELETE',
        url: path,
        responseType: 'json',
        data: payload,
      });
    }
}

export default new Request();
