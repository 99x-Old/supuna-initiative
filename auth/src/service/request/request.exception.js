export default class RequestException {
    name: string;

    message: string;

    status: number;

    stack: string;

    constructor(message: string, status: number = 500, data: {} = {}) {
      this.message = message;
      this.name = this.constructor.name;
      this.status = status;
      this.data = data;
    }
}
