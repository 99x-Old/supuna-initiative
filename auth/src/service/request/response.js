import type { ResponseType } from 'type/response.type';

export default (response: any) => {
  const data: ResponseType = {
    body: response.data ?? {},
    message: response.data.message,
  };
  return data;
};
