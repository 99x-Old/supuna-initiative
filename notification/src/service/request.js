import axios from 'axios';

export default class RequestService {
  post(url: string, data: any, headers: any = {
    'content-type': 'application/json',
  }): any {
    return axios({
      method: 'post',
      url,
      params: data,
      headers,
    });
  }

  get(url: string, data: any, headers: any): Promise<ResponseType> {
    return axios({
      method: 'get',
      url,
      params: data,
      headers,
    });
  }
}
