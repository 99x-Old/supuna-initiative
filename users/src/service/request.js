import axios from 'axios';

export default class RequestService {
  constructor() {}

  post(
    url: string,
    data: [],
    headers: any = {
      'content-type': 'application/json',
    },
  ) {
    return axios({
      method: 'post',
      url,
      params: data,
      headers,
    });
  }
}
