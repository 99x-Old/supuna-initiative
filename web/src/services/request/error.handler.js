import store from 'stores';
import { setAuth, setError } from 'actions';
import RequestException from './request.exception';
import type { ErrorResponseType } from '../../types/response.type';

export const redirectTo = (path: string) => {
  window.redirect = () => {
    window.location = path;
    return false;
  };
  window.redirect();
};

export default (error: ErrorResponseType) => {
  const { auth } = store.getState();

  if (error.response) {
    switch (error.response.status) {
      case 422:
        store.dispatch(setError(new RequestException(error.response
          .data.message, error.response.status, error.response.data.data)));
        break;
      case 403:
        break;
      case 401:
        store.dispatch(setError(
          new RequestException('You are not logged in!', error.response.status, {}),
        ));
        if (auth) {
          redirectTo('#auth-fail');
          store.dispatch(setAuth(null));
        }
        break;
      case 404:
        break;
      default:
        break;
    }
    return Promise.reject(
      new RequestException(
        error.response.data.message,
        error.response.status,
        error.response.data.data,
      ),
    );
  }
  return Promise.reject();
};
