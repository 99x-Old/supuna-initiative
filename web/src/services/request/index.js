import axios from 'axios';
import store from 'stores';
import type { HeaderType, ResponseType } from 'types/response.type';
import ErrorHandler from './error.handler';
import Response from './response';

class Request {
    contentType: string;

    headers: [];

    setDefault() {
      const service = axios.create({
        headers: this.getHeaders(),
      });
      service.interceptors.response.use(Response, ErrorHandler);
      this.service = service;
      this.contentType = 'application/x-www-form-urlencoded;charset=utf-8';
    }

    setHeaders(headers: []): Request {
      this.headers = headers;
      return this;
    }

    setContentType(contentType: string): Request {
      this.contentType = contentType;
      return this;
    }

    getHeaders(): HeaderType {
      const { auth } = store.getState();
      this.auth = auth;
      const headers = {
        'Content-Type': this.contentType,
      };

      if (auth) {
        headers.Authorization = `Bearer ${auth.access_token}`;
      }

      return {
        ...headers,
        ...this.headers,
      };
    }

    get(path: string, param: any): Promise<ResponseType> {
      this.setDefault();
      return this.service.get(path, param);
    }

    post(path: string, payload: any): Promise<ResponseType> {
      this.setDefault();
      return this.service.request({
        method: 'POST',
        url: path,
        responseType: 'json',
        data: payload,
      });
    }

    put(path: string, payload: any): Promise<ResponseType> {
      this.setDefault();
      return this.service.request({
        method: 'PUT',
        url: path,
        responseType: 'json',
        data: payload,
      });
    }

    delete(path: string, payload: any): Promise<ResponseType> {
      this.setDefault();
      return this.service.request({
        method: 'DELETE',
        url: path,
        responseType: 'json',
        data: payload,
      });
    }
}

export default new Request();
