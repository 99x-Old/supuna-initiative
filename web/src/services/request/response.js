import type { ResponseType } from 'types/response.type';

export default (response: any) => {
  const data: ResponseType = {
    body: response.data.data ?? {},
    message: response.data.message,
  };
  return data;
};
