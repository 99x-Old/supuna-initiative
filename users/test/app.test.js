import request from 'supertest';
import { test } from '@jest/globals';

const app = require('../app');

test('Health works', async () => {
  const response = await request(app.callback()).get('/health');
  expect(response.status).toBe(200);
  expect(response.body.status).toMatchSnapshot();
});
