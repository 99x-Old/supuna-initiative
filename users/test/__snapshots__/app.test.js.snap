// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Health works 1`] = `"working"`;
